From 4ee037f770b9e101d9fa00a63b1b20797b5bab85 Mon Sep 17 00:00:00 2001
From: Hoan Tran <hotran@apm.com>
Date: Wed, 14 Sep 2016 10:54:58 -0700
Subject: [PATCH 52/59] ACPI / CPPC: Support PCC with interrupt flag

Git-commit: b59c4b3dcd2e63d7a5483df3460d47278df716f7
Patch-mainline: v4.9-rc1
References: fate#319482

For PCC mailbox with interrupt flag, CPPC should call mbox_chan_txdone()
function to notify the mailbox framework about TX completion.

Signed-off-by: Hoan Tran <hotran@apm.com>
Reviewed-by: Prashanth Prakash <pprakash@codeaurora.org>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/acpi/cppc_acpi.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/acpi/cppc_acpi.c b/drivers/acpi/cppc_acpi.c
index 3d1ae6d37178..d0d0504b7c89 100644
--- a/drivers/acpi/cppc_acpi.c
+++ b/drivers/acpi/cppc_acpi.c
@@ -311,7 +311,10 @@ static int send_pcc_cmd(u16 cmd)
 	if (pcc_data.pcc_mrtt)
 		last_cmd_cmpl_time = ktime_get();
 
-	mbox_client_txdone(pcc_data.pcc_channel, ret);
+	if (pcc_data.pcc_channel->mbox->txdone_irq)
+		mbox_chan_txdone(pcc_data.pcc_channel, ret);
+	else
+		mbox_client_txdone(pcc_data.pcc_channel, ret);
 
 end:
 	if (cmd == CMD_WRITE) {
-- 
2.11.0

